src/forms/FormikDateField.stories.tsx

```diff
@@ -0,0 +1,188 @@
+import dayjs from 'dayjs'
+import utc from 'dayjs/plugin/utc'
+import { Form, Formik } from 'formik'
+import React from 'react'
+import * as yup from 'yup'
+
+import Button from '../Button'
+import FormikDateField from './FormikDateField'
+
+const dateSchema = yup.object().shape({
+  endDate: yup
+    .string()
+    .required('End date is required')
+    .test('check_minyear_1000', 'End date must be after year 1000', (value) => {
+      console.log(value, value?.substring(0, 4))
+      return Number(value?.substring(0, 4)) > 1000
+    })
+    .test(
+      'check_maxyear_9999',
+      'End date must be before year 9999',
+      (value) => {
+        console.log(
+          value,
+          value?.substring(0, 4),
+          Number(value?.substring(0, 4)),
+          Number(value?.substring(0, 5)),
+          isNaN(Number(value?.substring(0, 5)))
+        )
+        return (
+          isNaN(Number(value?.substring(0, 5))) ||
+          Number(value?.substring(0, 5)) < 9999
+        )
+      }
+    ),
+})
+
+export const Default = () => {
+  dayjs.extend(utc)
+
+  // A dayjs datestring in UTC time can be passed to the form for initialization and is parsed correctly
+  const initialDate = ''
+
+  return (
+    <div>
+      <div className="mb-4">
+        This field can be used for date handling, however keep in mind that the
+        date is stored as an HTML datetime-local string and not as UTC date
+        using dayjs. A possbile validation for this field and a possible submit
+        function converting the date to a corresponding object is implemented in
+        this example.
+      </div>
+      <Formik
+        initialValues={{
```

Comment:
Optional initial values passed to the component have to be transformed from zulu time in dayjs format to the datetime format of the html input component